timeboost-sequencer/src/decrypt.rs

```diff
@@ -442,23 +493,37 @@ impl Worker {
                     DecryptedItem(Bytes::from(<Vec<u8>>::from(decrypted_data))),
                 ))
             })
-            .collect::<Result<_, DecryptError>>()?;
+            .collect::<Result<_>>()?;
 
         // clean up
         self.shares.retain(|k, _| k.round() != round);
         for cid in to_remove {
-            self.idx2cid
-                .remove_by_right(&cid)
+            self.cid2idx
+                .remove(&cid)
                 .ok_or(DecryptError::MissingIndex(cid))?;
             self.cid2ct
-                .remove_by_left(&cid)
+                .remove_by_left(&(round, cid))
                 .ok_or(DecryptError::MissingCiphertext(cid))?;
         }
 
         Ok(Some((round, decrypted.into_values().collect())))
     }
 }
 
+/// Serialize a given data type into `Bytes`
+fn serialize<T: Serialize>(d: &T) -> Result<bytes::Bytes> {
+    let mut b = BytesMut::new().writer();
+    bincode::serde::encode_into_std_write(d, &mut b, bincode::config::standard())?;
+    Ok(b.into_inner().freeze())
```

Comment:
```suggestion
    Ok(b.into_inner())
```
If `serialize` returns `BytesMut` it could be passed directly to the network.

Reply:
nice, thanks.